redhat-cop/monitoring
# File based sd, inventory overhaul

### What does this PR do?
This PR changes the way how prometheus targets are configured. The PR adds file_based_sd on all of the targets.
The targets are now created by running the add-target role.

Also the inventory has been changed to use groups/hosts instead of plain vars on prometheus-scraper. Example inventory has been updated accordingly

The error handling on configure-grafana-datasource has been configured to continue when datasource is already exists.


### How should this be tested?
Update the example inventory with correct hosts. edit host_vars with relevant data for your openshift environment.
 playbooks/infra-prometheus/inventory/host_vars/openshift-1.yml. 
Run setup-prometheus-grafana.yml playbook
Run configure-grafana-datasource.yml playbook
Run add-targets.yml playbook

After these four steps you should have running and configured prometheus,grafana,alertmanager.

### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.
resolves #<number>

### Other Relevant info, PRs, etc.
Please provide link to other PRs that may be related (blocking, resolves, etc. etc.)

### People to notify
cc: @redhat-cop/monitoring
